File: GameLibrary/src/cc/lib/swing/AWTEditText.java
package cc.lib.swing;

import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public abstract class AWTEditText extends JTextField implements KeyListener {

    public AWTEditText(String text, int maxLength) {
        super(text, maxLength);
        addKeyListener(this);
    }

    @Override
    public final void keyTyped(KeyEvent e) {
        onTextChanged(getText());
    }

    @Override
    public final void keyPressed(KeyEvent e) {

    }

    @Override
    public final void keyReleased(KeyEvent e) {
        onTextChanged(getText());
    }

    protected abstract void onTextChanged(String newText);
}

File: SwingApps/src/cc/lib/swing/AWTEditText.kt
package cc.lib.swing

import java.awt.event.KeyEvent
import java.awt.event.KeyListener
import javax.swing.JTextField

abstract class AWTEditText(text: String?, maxLength: Int) : JTextField(text, maxLength), KeyListener {
	init {
		addKeyListener(this)
	}


	override fun keyPressed(e: KeyEvent) {}
	override fun keyReleased(p0: KeyEvent?) {}
	override fun keyTyped(e: KeyEvent) {
		if (e.keyCode == KeyEvent.VK_ENTER) {
			text?.let {
				onReturnKey(text)
			}
		}
	}

	protected abstract fun onReturnKey(newText: String)
}